app/_hub/kong-inc/opa/index.md

```diff
@@ -0,0 +1,307 @@
+      "created_at": 1612819937,
+      "connect_timeout": 60000,
+      "id": "e6fd8b19-89e5-44e6-8a2a-79e8bf3c31a5",
+      "protocol": "http",
+      "name": "foo",
+      "read_timeout": 60000,
+      "port": 80,
+      "updated_at": 1612819937,
+      "ws_id": "d6020dc4-67f5-4c62-8b45-e2f497c20f5c",
+      "retries": 5,
+      "write_timeout": 60000
+    },
+    "route": {               # The Kong route that was matched for this request. Injected only if `include_route_in_opa_input` is set to `true`.
+      "id": "bc6d8617-76a7-441f-aa40-32eb1f5be9e6",
+      "paths": [
+        "\\/"
+      ],
+      "protocols": [
+        "http",
+        "https"
+      ],
+      "strip_path": true,
+      "created_at": 1612819949,
+      "ws_id": "d6020dc4-67f5-4c62-8b45-e2f497c20f5c",
+      "request_buffering": true,
+      "updated_at": 1612819949,
+      "preserve_host": false,
+      "regex_priority": 0,
+      "response_buffering": true,
+      "https_redirect_status_code": 426,
+      "path_handling": "v0",
+      "service": {
+        "id": "e6fd8b19-89e5-44e6-8a2a-79e8bf3c31a5"
+      }
+    },
+    "consumer": {            # Kong consumer that was used for authentication for this request. Injected only if `include_consumer_in_opa_input` is set to `true`.
+      "id": "bc6d8617-76a7-431f-aa40-32eb1f5be7e6",
+      "username": "kong-consumer-username"
+    }
+  }
+}
+
+```
+
+> Please note that it is possible that Consumer and Service resources are not present for any given request in Kong.
+
+## Expected response from OPA
+
+Once OPA is done executing policies, the plugin expects the policy evaluation
+result in one of the two defined formats. Any other format will result in the
+plugin returning a `500 Internal Server Error` to the client.
+If the HTTP status code of the response from OPA is not `200 OK`, then it is
+treated as an error, which results in a `500 Internal Server Error` to the client.
+
+### Boolean result
+
+OPA can return a `true` or `false` result after a policy evaluation.
+This should suffice for most use cases.
+
+Examples:
+
+To allow the request to be processed further, OPA should send the following
+response:
+
+```json
+{
+  "result": true
+}
+```
+
+To deny the request and terminate its processing, OPA should send the following
+response:
+
+```json
+{
+  "result": false
+}
+```
+
+Any other fields in the response in this case are ignored.
+
+### Object result
+
+In this case, OPA returns an object instead of a boolean result. This
+allows for injecting custom HTTP headers as well as changing the HTTP status code
+for rejected requests.
+
+The plugin expects the following structure in the OPA response in this case:
+
+```json
+{
+  "result": {
+    "allow": <boolean>,
+    "status: <HTTP status code>,
+    "headers": {
+      "<key>": "<value>",
+      "<key2>": "<value2>"
+    }
+  }
+}
+```
+
+The only required field in this response is `result.allow` and it must be a
+`boolean`, or else the plugin will error out.
```

Comment:
```suggestion
The only required field in this response is `result.allow`, which accepts a
`boolean` value.
```
Saying something like "or else this will fail" is really unnecessary - this is implied by the construction. Otherwise, we'd be saying "or else failure" everywhere that we define required parameters.